src/main/java/bookstore/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,33 @@
+package bookstore.repository.book;
+
+import bookstore.dto.BookSearchParameters;
+import bookstore.model.Book;
+import bookstore.repository.SpecificationBuilder;
+import bookstore.repository.SpecificationProviderManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParameters searchParameters) {
+        Specification<Book> spec = Specification.where(null);
+        if (searchParameters.authors() != null && searchParameters.authors().length > 0) {
+            spec = spec.and(bookSpecificationProviderManager.getSpecificationProvider("author")
+                    .getSpecification(searchParameters.authors()));
+        }
+        if (searchParameters.titles() != null && searchParameters.titles().length > 0) {
+            spec = spec.and(bookSpecificationProviderManager.getSpecificationProvider("title")
+                    .getSpecification(searchParameters.titles()));
+        }
+        if (searchParameters.prices() != null && searchParameters.prices().length > 0) {
+            spec = spec.and(bookSpecificationProviderManager.getSpecificationProvider("price")
+                    .getSpecification(searchParameters.prices()));
```

Comment:
I guess we can split it into several methods for better code readability. Also, we can use Optional for the null checks and streamline the code

Reply:
Sorry, but i think that i am not able to do this without someone's help(((